cookbooks/continuous_delivery.rst

```diff
@@ -0,0 +1,245 @@
+use of a "Machine user":  head to `SymfonyConnect <https://connect.symfony.com>`_
+and register a new dedicated account.
+
+As this context will not be interactive, interactive authentication will not be
+possible, this is why you have to use an :doc:`API token
+</cookbooks/api_tokens>`.
+
+Additionally, as deployment are based on Git, you will
+have to generate a new SSH key, and register it in SymfonyCloud:
+
+.. code-block:: terminal
+
+   $ ssh-keygen -N '' -f symfonycloud-ci.key
+   $ SYMFONY_TOKEN=xxxx symfony account:ssh:key:add symfonycloud-ci.key
+
+.. caution::
+
+   Don't forget to register and expose the ``SYMFONY_TOKEN`` environment value
+   as well as the generated SSH key in your automation system.
+
+When running ``symfony deploy``, SSH host keys are checked to ensure not MITM
+attack is happening. This security measure requires you to register the
+following SSH host key as a know key in your automation system (usually in
+``~/.ssh/known_hosts``):
+
+.. literalinclude:: ../_includes/ssh-host-key-eu.txt
+  :language: text
+
+.. caution::
+
+   Disabling SSH hosts key checking or using ``ssh-keyscan`` in your automation
+   pipeline to register the SSH host key breach several security measures and
+   should not be done.
+
+.. tip::
+
+   Even if this is unlikely, this key might change in the future. You can use
+   ``ssh-keyscan git.eu.s5y.io`` to ensure its freshness.
+
+CLI behavior
+^^^^^^^^^^^^
+
+When the CLI tool detects it is currently running in an automation tool,
+interaction is disabled. If you run a command that requires some input, default
+values are going to be used.
+
+In order to determine this the CLI checks the following:
+
+* stdin is a terminal or not
+* ``DEBIAN_FRONTEND`` environment variable is set to ``noninteractive``
+* Presence of several environment variable based on commonly used by CI systems
+  (``BUILD_ID``, ``GITLAB_CI``, ``TRAVIS``, ``CIRCLECI``, etc)
+* ``SHELL_INTERACTIVE`` environment variables is set to ``''`` (the empty
+  string)
+
+If your system is not automatically detected, you can use the
+``--no-interaction`` flag.
+
+.. _cd-env-flags:
+
+CLI flags and environment variables
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The CLI tool exposes flags and check some environment variables in order to
+allow you to change its behavior. As a general rule, for automated context,
+environment variables are preferable. Flags can be used to override environment
+variables punctually:
+
+.. code-block:: terminal
+
+   $ export SYMFONY_ENVIRONMENT=pr-1
+   $ symfony deploy
+   # Deploying to pr-1
+   $ symfony deploy --environment=staging
+   # Deploying to staging
+
+SYMFONY_PROJECT
+"""""""""""""""
+
+Setting this environment value replace the need to run ``symfony project:link``.
+
+Its value should be set to the SymfonyCloud project's id that can be found with
+``symfony project:info`` or in the emails sent by SymfonyCloud.
+
+.. tip:: One can also use the ``--project`` flag.
```

Comment:
Deploying with `--project` works for me; setting `SYMFONY_PROJECT` as an environment variable doesn't.

Reply:
Fixed in v4.18.4